lib/project_types/script/layers/application/push_script.rb

```diff
@@ -14,24 +14,23 @@ def call(ctx:, force:, project:)
 
             extension_point = ExtensionPoints.get(type: script_project.extension_point_type)
             library_name = extension_point.libraries.for(script_project.language)&.package
-            raise Infrastructure::Errors::LanguageLibraryForAPINotFoundError.new(
-              language: script_project.language,
-              api: script_project.extension_point_type
-            ) unless library_name
```

Comment:
Hmm I'm guessing that this isn't dangerous to remove? I'm trying to think if what would happen if non-wasm project is misconfigured. As long as the `package` is specified in the extension point config, this should always return a value right?

Also if we still want to remove this, we'll also have to clean up the error message for it.

Reply:
With the shift to Wasm, we're going to stop collecting/showing the library version, right @lopert? If that's true, then it might be fair to remove this since we wouldn't have a need for it.

If not, we still need to support not having a library for Wasm projects. I would suggest that instead of trying to catch a misconfigured extension point here, we should set up a test in the CLI repo to catch invalid values when they're set by first-party devs. In this case, this runtime check shouldn't be needed anymore.

WDYT?

Reply:
This was added back when we were 1p library only, so we always wanted to forward the library information. Now that we support wasm, we have a situation where we won't have a library, so this check isn't really valid anymore.

I don't see anything dangerous by removing this. 
To verify, I tested passing in a nil library_name to the library object. It ends up being rejected by script_service.

As for how we'd get into this state, I think the scenario we had envisioned was if a language was no longer supported (ex: after a major protocol change). An old script could exist, and trying to push it with an updated CLI would raise the error.

I think the way forward for developers in this situation would be to turn their script project into a `wasm` project, handle the build steps themselves, and then push their modules up as `wasm`.

Reply:
To clarify:
- I think we should keep the library information in for now. I spoke to @smacleodify briefly about this and we figured that keeping it in could be nice for data, and it's already implemented.
- I think the situation where this error would occur is pretty narrow, but I also don't mind keeping it in? I could add a conditional here that raises the error, unless the `script_project.language == "wasm"`

Reply:
Ok after thinking on it, I settled on:

1. Keeping the library data.
2. Keeping the error message, but adding a small conditional for "wasm". This is the only spot we treat Wasm differently.
  a. This allows us to return the better error and fail locally instead of waiting to be rejected by script-service
3.  Added tests to cover the wasm interactions.